### DISCUSSION TITLE
Markdown shortcuts: Trigger a code block with ``` without a space?

### DISCUSSION THREAD

**Discussion by Nic0S** [facebook/lexical#6712] [Q&A]
In v0.17.0, it was possible to trigger a code block with three backticks (no space at the end) by modifying the built in CODE transformer like this:

```
  { ...CODE, regExp: /^```/ },
```

v0.18.0 changed the CODE transformer: https://github.com/facebook/lexical/pull/6530/files#diff-bc77b598e2f79b614dd555952e1c10dab948dcfddfb11d7c6583235fa3e60815L81

So I think the equivalent would be:
```
  {
    ...CODE,
    regExpStart: /^```/,
    regExpEnd: {
      optional: true,
      regExp: /```$/
    }
  }
```

But this no longer triggers the code block without a space after the backticks. Another solution I've tried is just using the pre-v0.18.0 version of the CODE transformer with the regExp override, like this:

```
  {
    dependencies: [CodeNode],
    export: (node: LexicalNode) => {
      if (!$isCodeNode(node)) {
        return null;
      }
      const textContent = node.getTextContent();
      return (
        "```" + (node.getLanguage() || "") + (textContent ? "\n" + textContent : "") + "\n" + "```"
      );
    },
    regExp: /^```/,
    replace: createBlockNode((match) => {
      return $createCodeNode(match ? match[1] : undefined);
    }),
    type: "element",
  }
```

But even this won't work without a space after the ``` backticks.

Has the underlying MarkdownShortcutPlugin behavior changed in a way that prevents triggering this without a space?

**Comment by romancavry** [comment]
@Nic0S Hello, did you manage this?

**Reply by Nic0S** [reply]
I did not, no longer on the team working on this but at the time we gave up on the upgrade
